src/ssmDocument/wizards/publishDocumentWizard.ts

```diff
@@ -3,400 +3,140 @@
-    private documents: SSM.Types.DocumentIdentifierList | undefined
-
-    private readonly totalSteps: number = 3
+export class PublishSSMDocumentWizard extends Wizard<PublishSSMDocumentWizardResponse> {
+    public constructor(region?: string) {
+        super({ initState: { region } })
+        const form = this.form
 
-    public constructor(private readonly awsContext: AwsContext, private readonly regionProvider: RegionProvider) {
-        super()
-    }
-
-    public async promptUserForRegion(initialRegionCode?: string): Promise<string | undefined> {
-        const partitionRegions = getRegionsForActiveCredentials(this.awsContext, this.regionProvider)
+        // TODO: add prompt when more types are supported
+        form.documentType.setDefault('Automation')
 
-        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
-            options: {
+        form.region.bindPrompter(() =>
+            createRegionPrompter(undefined, {
                 title: localize(
                     'AWS.message.prompt.ssmDocument.publishDocument.region',
                     'Which AWS Region would you like to publish to?'
                 ),
-                value: initialRegionCode,
-                matchOnDetail: true,
-                ignoreFocusOut: true,
-                step: 1,
-                totalSteps: this.totalSteps,
-            },
-            items: partitionRegions.map(region => ({
-                label: region.name,
-                detail: region.id,
-                // this is the only way to get this to show on going back
-                // this will make it so it always shows even when searching for something else
-                alwaysShow: region.id === initialRegionCode,
-                description: region.id === initialRegionCode ? recentlyUsed : '',
-            })),
-            buttons: [vscode.QuickInputButtons.Back],
-        })
-
-        const choices = await picker.promptUser<vscode.QuickPickItem>({
-            picker: quickPick,
-            onDidTriggerButton: (button, resolve, reject) => {
-                if (button === vscode.QuickInputButtons.Back) {
-                    resolve(undefined)
-                }
-            },
-        })
-        const val = picker.verifySinglePickerOutput(choices)
-
-        return val?.detail
-    }
-
-    public async loadSSMDocument(region: string, documentType?: SSM.Types.DocumentType): Promise<void> {
-        if (!this.documents) {
-            const filters: SSM.Types.DocumentKeyValuesFilterList = [
-                {
-                    Key: 'Owner',
-                    Values: ['Self'],
-                },
-            ]
-            if (documentType !== undefined) {
-                filters.push({
-                    Key: 'DocumentType',
-                    Values: [documentType],
-                })
+                serviceFilter: 'ssm',
```

Comment:
We should be applying service filters to relevant prompts. Many services aren't available in every region.

Reply:
The service filter does mean we should keep the Toolkit/SDK in-sync in terms of endpoints which we don't do.
Same applies to the explorer tree. We release enough that I don't think it's much of a problem though.